mlflow_databricks_artifacts/store/artifact_repo.py

```diff
@@ -0,0 +1,323 @@
+import base64
+import logging
+import os
+import posixpath
+import requests
+import uuid
+
+from mlflow import get_tracking_uri
+from mlflow.entities import FileInfo
+from mlflow.exceptions import MlflowException
+from mlflow.store import artifact_repo
+from mlflow.protos.service_pb2 import MlflowService, GetRun, ListArtifacts
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, INTERNAL_ERROR
+
+try:
+    from mlflow.protos.databricks_artifacts_pb2 import (
+        DatabricksMlflowArtifactsService,
+        GetCredentialsForWrite,
+        GetCredentialsForRead,
+        ArtifactCredentialType,
+    )
+except Exception:
+    from mlflow_databricks_artifacts.protos.databricks_artifacts_pb2 import (
+        DatabricksMlflowArtifactsService,
+        GetCredentialsForWrite,
+        GetCredentialsForRead,
+        ArtifactCredentialType,
+    )
+
+from mlflow_databricks_artifacts.azure.client import put_block, put_block_list
+from mlflow_databricks_artifacts.utils.databricks_utils import get_databricks_host_creds
+from mlflow_databricks_artifacts.utils.file_utils import relative_path_to_artifact_path, yield_file_in_chunks
+from mlflow_databricks_artifacts.utils.proto_json_utils import message_to_json
+from mlflow_databricks_artifacts.utils.rest_utils import (
+    call_mlflow_endpoint,
+    extract_api_info_for_service,
+    http_put,
+    http_get,
+)
+from mlflow_databricks_artifacts.utils.uri import (
+    extract_and_normalize_path,
+    get_databricks_profile_uri_from_artifact_uri,
+    is_databricks_acled_artifacts_uri,
+    is_valid_dbfs_uri,
+    remove_databricks_profile_info_from_artifact_uri,
+)
+
+_logger = logging.getLogger(__name__)
+_PATH_PREFIX = "/api/2.0"
+_AZURE_MAX_BLOCK_CHUNK_SIZE = 100000000  # Max. size of each block allowed is 100 MB in stage_block
+_DOWNLOAD_CHUNK_SIZE = 100000000
+_SERVICE_AND_METHOD_TO_INFO = {
+    service: extract_api_info_for_service(service, _PATH_PREFIX)
+    for service in [MlflowService, DatabricksMlflowArtifactsService]
+}
+
+
+class DatabricksArtifactRepository(artifact_repo.ArtifactRepository):
```

Comment:
This definition should be almost identical to https://github.com/mlflow/mlflow/blob/f843d044dda241f31d736f48263379bfc7e963f5/mlflow/store/artifact/databricks_artifact_repo.py#L48, with a couple of changes:

1. Instead of leveraging `azure-storage-blob` for uploads, this artifact repo leverages new utility methods (`mlflow_databricks_artifacts/azure/client.py`)

2. This artifact repository does not handle artifact listing pagination due to aforementioned difficulties with re-defining / re-including MLflow service protos (which contain the `ListArtifacts` API definition) in this project.

Reply:
Neat